frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditHeader.tsx

```diff
@@ -1,71 +1,90 @@
-import { useMemo } from 'react'
+import { useEffect, useMemo } from 'react'
 import { useForm } from 'react-hook-form'
 import { useDebounce } from 'react-use'
 import { Divider, FormControl, Stack } from '@chakra-ui/react'
 import { extend } from 'lodash'
 
+import { SectionFieldBase } from '~shared/types/field'
+
 import { createBaseValidationRules } from '~utils/fieldValidation'
 import FormErrorMessage from '~components/FormControl/FormErrorMessage'
 import FormLabel from '~components/FormControl/FormLabel'
 import Input from '~components/Input'
 import Textarea from '~components/Textarea'
-import { SectionFieldSchema } from '~templates/Field/Section/SectionFieldContainer'
-
-import { useEditFieldStore } from '../../../editFieldStore'
-import { useMutateFormFields } from '../../../mutations'
-import { isPendingFormField } from '../../../utils'
 
-import { DrawerContentContainer } from './DrawerContentContainer'
-import { FormFieldDrawerActions } from './FormFieldDrawerActions'
+import { DrawerContentContainer } from './common/DrawerContentContainer'
+import { FormFieldDrawerActions } from './common/FormFieldDrawerActions'
+import { FieldMutateOptions } from './common/types'
 
 export interface EditHeaderProps {
-  field: SectionFieldSchema
+  field: SectionFieldBase
+  isLoading: boolean
+  isPendingField: boolean
+  handleChange: (field: SectionFieldBase) => void
+  handleSave: (field: SectionFieldBase, options?: FieldMutateOptions) => void
+  handleCancel: () => void
 }
 
 interface EditHeaderInputs {
   title: string
   description: string
 }
 
-export const EditHeader = ({ field }: EditHeaderProps): JSX.Element => {
-  const { updateActiveField, clearActiveField } = useEditFieldStore()
-
+export const EditHeader = ({
+  field,
+  isLoading,
+  isPendingField,
+  handleChange,
+  handleSave,
+  handleCancel,
+}: EditHeaderProps): JSX.Element => {
   const {
     handleSubmit,
-    reset,
     watch,
     register,
     formState: { errors, isDirty },
+    reset,
+    setValue,
   } = useForm<EditHeaderInputs>({
     defaultValues: {
       title: field.title,
       description: field.description,
     },
   })
 
-  const watchedInputs = watch()
+  // Update form when field loads or changes due to external action,
+  // e.g. if user clicks on another field in the builder
```

Comment:
I don't think so actually. if we unmount when the fieldId changes then the component will unmount and remount when a newly created field is changed, because the fieldId will change from the pending ID to the field's actual ID. then the `onSuccess` or `onFailure` handlers for the don't fire because the component is unmounted when the mutation completes.

Reply:
Hmm, can we update the onSuccess handler of the createField query to update the fieldId instead of doing it earlier? That would allow the onSuccess to run before the id is updated

Reply:
but still a bit janky right - if we add an `onSuccess` on the component in future, it will fail